packages/widget.vite-plugin.unstable/src/build-plugin/extractBuildOutputs.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2025 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from "fs";
+import type { DefaultTreeAdapterTypes } from "parse5";
+import { parse as parse5 } from "parse5";
+
+export interface BuildOutputs {
+  scripts: Array<ScriptBuildOutput>;
+  stylesheets: string[];
+}
+
+type BuildOutput = ScriptBuildOutput | StylesheetBuildOutput;
+
+type ScriptType = "text/javascript" | "module";
+type ScriptBuildOutput = {
+  type: "script";
+  scriptType: ScriptType;
+  src: string;
+};
+
+type StylesheetBuildOutput = {
+  type: "stylesheet";
+  src: string;
+};
+
+export function extractBuildOutputs(
+  htmlFilePath: string,
+): BuildOutputs {
+  const htmlContent = fs.readFileSync(htmlFilePath, "utf-8");
+  return parseTransformResult(htmlContent);
+}
+
+/**
+ * Parse the HTML transform result and extract the injected script details.
+ */
+function parseTransformResult(result: string): BuildOutputs {
+  const outputs = visitNode(parse5(result));
+  return {
+    scripts: outputs.filter((output) =>
+      output.type === "script"
+    ) as ScriptBuildOutput[],
+    stylesheets: outputs.filter((output) => output.type === "stylesheet").map((
+      output,
+    ) => output.src),
+  };
+}
+
+function visitNode(node: DefaultTreeAdapterTypes.Node): BuildOutput[] {
+  if (node.nodeName === "script") {
+    return [parseScriptNode(node as DefaultTreeAdapterTypes.Element)];
+  }
+  if (node.nodeName === "link" && getAttribute(node, "rel") === "stylesheet") {
+    return [parseStylesheetNode(node as DefaultTreeAdapterTypes.Element)];
+  }
+  if ("childNodes" in node) {
+    return node.childNodes.flatMap((childNode) => visitNode(childNode));
+  }
+  return [];
+}
+
+function parseScriptNode(
```

Comment:
Should we be stricter on this to not allow other script type attribute values and other script attributes? I'm not sure if it's bad to break on the assumption of what tag structure vite will inject into the html but also at the moment we would let through a lot of stuff perhaps unintentionally:

- Only allow script tag if type attribute is module, javascript mime type or empty??
- Only allow src attribute and no others
- Disallow any text content inside the tag